src/app/pages/dashboard/components/widget-group-form/widget-group-form.component.ts

```diff
@@ -1,47 +1,161 @@
-import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
-import { FormBuilder } from '@angular/forms';
+import {
+  ChangeDetectionStrategy, ChangeDetectorRef, Component,
+} from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
 import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
-import { of } from 'rxjs';
+import {
+  Observable, filter, of, tap,
+} from 'rxjs';
+import { Option } from 'app/interfaces/option.interface';
 import { ChainedRef } from 'app/modules/ix-forms/components/ix-slide-in/chained-component-ref';
-import { WidgetGroup, WidgetGroupLayout, widgetGroupIcons } from 'app/pages/dashboard/types/widget-group.interface';
+import { WidgetGroupFormStore } from 'app/pages/dashboard/components/widget-group-form/widget-group-form.store';
+import { WidgetCategory } from 'app/pages/dashboard/types/widget-category.enum';
+import {
+  WidgetGroup, WidgetGroupLayout, layoutToSlotSizes, widgetGroupIcons,
+} from 'app/pages/dashboard/types/widget-group.interface';
+import { WidgetType } from 'app/pages/dashboard/types/widget.interface';
+import { widgetRegistry } from 'app/pages/dashboard/widgets/all-widgets.constant';
 
 @UntilDestroy()
 @Component({
   selector: 'ix-widget-group-form',
   templateUrl: './widget-group-form.component.html',
   styleUrls: ['./widget-group-form.component.scss'],
+  providers: [
+    WidgetGroupFormStore,
+  ],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class WidgetGroupFormComponent {
   protected group: WidgetGroup;
   protected selectedSlot = 0;
-  protected form = this.formBuilder.group({
-    template: [''],
-    layout: [WidgetGroupLayout.Full],
-  });
-  readonly layoutsMap = widgetGroupIcons;
+
+  protected readonly template = new FormControl('');
```

Comment:
I think component store makes your life harder here, because instead of having one source of truth you things stored in three places. There are form controls, there's `group` inside of the component and there's a store.
It's really hard to trace what changes what here. 

Having a separate place for logic is a good idea, but then we should come up with a way to utilize it more.

We should avoid doing what we did in pool creation wizard, because it really makes for some complicated logic. 

Options for dropdowns should react to changes in data store whereever it is.

See this video https://www.youtube.com/watch?v=skOTEbGwncE
It focuses on rxjs, but you could do the same thing easier with signals.

You need to pick one place for source of truth (could be a store, could be a form or signals here or in a service) and then have other things derive from it.